ocaml/xapi/bios_strings.ml

```diff
@@ -23,6 +23,50 @@ let remove_invisible str =
   let str = String.concat "\n" l in
   String.fold_left (fun s c -> if c >= ' ' && c <= '~' then s ^ (String.of_char c) else s) "" str
 
+(* A single record from the output of dmidecode,
+ * without its type, handle nor size *)
+type record = {name: string; values: (string * string) list}
+
+(* Parser module for the output of command dmidecode -q *)
+module P = struct
+  open Angstrom
+  let is_eol =   function | '\n' -> true | _ -> false
+  let is_tab =   function | '\t' -> true | _ -> false
+  let is_space = function | ' '  -> true | _ -> false
+  let is_sep =   function | ':'  -> true | _ -> false
+
+  let single_sep = skip is_sep *> skip is_space
+  let multi_sep = skip is_sep *> end_of_line
+
+  let token = take_till is_eol <* end_of_line
+  let key = skip is_tab *> take_till is_sep
+  let v_line = skip is_tab *> skip is_tab *> token
+
+  let single_line =
+    lift2 (fun k v -> k, v)
+    (key <* single_sep) (* k *)
+    (token) (* v *)
+
+  let multi_line =
+    lift2 (fun k v_lines -> k, String.concat "\n" v_lines)
```

Comment:
Looks good but IDK how general we want to be cause there can be some output that does not match either of these e.g.
```
BIOS Language Information
        Language Description Format: Long
        Installable Languages: 2
                en|US|iso8859-1
                <BAD INDEX>
        Currently Installed Language: en|US|iso8859-1
```
But if we will never encounter these then it's fine

Reply:
Did the current code give problems regarding malformed output? I think that logging a warning in the case that the call to dmidecode fails or parsing it should be enough.

Reply:
This is not malformed, but when I get the entire output with my dmidecode, it displays stuff like this towards the end.

Reply:
I misunderstood, I can add that case (I hadn't seen that array output in any of the machines)

Reply:
Now the parser supports this format, the end result is a single string with endlines separating the different values. I think we can get away with this since we were not processing these kinds of values before.